Lib/dataclasses.py

```diff
@@ -0,0 +1,776 @@
+        if type(a_type) is typing._ClassVar:
+            # This field is a ClassVar, so it's not a field.
+            f._field_type = _FIELD_CLASSVAR
+
+    if f._field_type is _FIELD:
+        # Check if this is an InitVar.
+        if a_type is InitVar:
+            # InitVars are not fields, either.
+            f._field_type = _FIELD_INITVAR
+
+    # Validations for fields.  This is delayed until now, instead of
+    # in the Field() constructor, since only here do we know the field
+    # name, which allows better error reporting.
+
+    # Special restrictions for ClassVar and InitVar.
+    if f._field_type in (_FIELD_CLASSVAR, _FIELD_INITVAR):
+        if f.default_factory is not _MISSING:
+            raise TypeError(f'field {f.name} cannot have a '
+                            'default factory')
+        # Should I check for other field settings? default_factory
+        #  seems the most serious to check for. Maybe add others.  For
+        #  example, how about init=False (or really,
+        #  init=<not-the-default-init-value>)? It makes no sense for
+        #  ClassVar and InitVar to specify init=<anything>.
+
+    # For real fields, disallow mutable defaults for known types.
+    if f._field_type is _FIELD and isinstance(f.default, (list, dict, set)):
+        raise ValueError(f'mutable default {type(f.default)} for field '
+                         f'{f.name} is not allowed: use default_factory')
+
+    return f
+
+
+def _find_fields(cls):
+    # Return a list of Field objects, in order, for this class (and no
+    #  base classes).  Fields are found from __annotations__ (which is
+    #  guaranteed to be ordered).  Default values are from class
+    #  attributes, if a field has a default.  If the default value is
+    #  a Field(), then it contains additional info beyond (and
+    #  possibly including) the actual default value.  Pseudo-fields
+    #  ClassVars and InitVars are included, despite the fact that
+    #  they're not real fields.  That's deal with later.
+
+    annotations = getattr(cls, '__annotations__', {})
+
+    return [_get_field(cls, a_name, a_type)
+            for a_name, a_type in annotations.items()]
+
+
+def _set_attribute(cls, name, value):
+    # Raise TypeError if an attribute by this name already exists.
+    if name in cls.__dict__:
+        raise TypeError(f'Cannot overwrite attribute {name} '
+                        f'in {cls.__name__}')
+    setattr(cls, name, value)
+
+
+def _process_class(cls, repr, eq, order, hash, init, frozen):
+    # Use an OrderedDict because:
+    #  - Order matters!
+    #  - Derived class fields overwrite base class fields, but the
+    #    order is defined by the base class, which is found first.
+    fields = collections.OrderedDict()
+
+    # Find our base classes in reverse MRO order, and exclude
+    #  ourselves.  In reversed order so that more derived classes
+    #  override earlier field definitions in base classes.
+    for b in cls.__mro__[-1:0:-1]:
```

Comment:
Not `mro()`?

Reply:
How would that be an improvement?

Reply:
I don't know at what stage this code is executed. But if there is a chance for a difference between `mro()` and `__mro__` (for example if it is possible that `__mro__` is still not initialized), there may be a reason of using `mro()`.